scripts/imgtool/main.py

```diff
@@ -477,10 +485,46 @@ def sign(key, public_key_format, align, version, pad_sig, header_size,
         }
 
     img.create(key, public_key_format, enckey, dependencies, boot_record,
-               custom_tlvs, int(encrypt_keylen), clear, baked_signature,
+               custom_tlvs, compression_tlvs, int(encrypt_keylen), clear, baked_signature,
                pub_key, vector_to_sign)
-    img.save(outfile, hex_addr)
 
+    if re.match('.*\.bin$', infile) is not None :
+        if compression == "lzma2" :
+            compressed_img = image.Image(version=decode_version(version), header_size=header_size,
+                      pad_header=pad_header, pad=pad, confirm=confirm,
+                      align=int(align), slot_size=slot_size,
+                      max_sectors=max_sectors, overwrite_only=overwrite_only,
+                      endian=endian, load_addr=load_addr, rom_fixed=rom_fixed,
+                      erased_val=erased_val, save_enctlv=save_enctlv,
+                      security_counter=security_counter, max_align=max_align)
+            compressed_file =  re.sub('zephyr\.bin$','zephyr.compressed.bin',infile)
+            compression_filters = [
+                {"id": lzma.FILTER_LZMA1, "dict_size": 131072},
+                {"id": lzma.FILTER_LZMA1, "lp": 1},
+                {"id": lzma.FILTER_LZMA1, "lc": 3},
+                {"id": lzma.FILTER_LZMA1, "preset": 9},
```

Comment:
lzma2?

Reply:
Tried in the first place and it also didn't work. I will look into down the line.

Reply:
Also other quote:
> When opening a file for writing, the settings used by the
        compressor can be specified either as a preset compression
        level (with the *preset* argument), or in detail as a custom
        filter chain (with the *filters* argument). For FORMAT_XZ and
        FORMAT_ALONE, the default is to use the PRESET_DEFAULT preset
        level. For FORMAT_RAW, the caller must always specify a filter
        chain; the raw compressor does not support preset compression
        levels.